### PR TITLE
Use namedMain and patch vendor lib

### PR DESCRIPTION
**Pull request by matzew**
Signed-off-by: Matthias Wessendorf <mwessend@redhat.com>

a  little more tweaking on the vendor'ed dependency

### CODE REVIEW

File: vendor/knative.dev/pkg/injection/sharedmain/main.go [hunk 1]
```diff
@@ -120,12 +119,15 @@ var (
 // by name.
 func MainNamed(ctx context.Context, component string, ctors ...injection.NamedControllerConstructor) {
 
-	disabledControllers := pflag.StringSlice("disable-controllers", []string{}, "Comma-separated list of disabled controllers.")
+	var (
+		disabledControllers string
+	)
+	flag.StringVar(&disabledControllers, "disable-controllers", "", "Comma-separated list of disabled controllers.")
```
Comment (pierDipi) [inline vendor/knative.dev/pkg/injection/sharedmain/main.go:125]:
```suggestion
	disabledControllers := flag.String("disable-controllers", "", "Comma-separated list of disabled controllers.")
```

Comment (matzew) [reply vendor/knative.dev/pkg/injection/sharedmain/main.go:125]: I think most eventing does use the `flag.StringVar` - that's why :man_shrugging:

Comment (pierDipi) [reply vendor/knative.dev/pkg/injection/sharedmain/main.go:125]: /lgtm

**Pull request merged**